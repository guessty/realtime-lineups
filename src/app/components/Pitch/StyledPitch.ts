import styled from 'styled-components'

export default styled.div`
  text-align: center;
  background: url('assets/pitch-bg.png');
  padding: 2rem;

  > div {
    align-items: center;
  }

  @media(max-width: 600px) {
    padding: 2rem 0;
  }
`
