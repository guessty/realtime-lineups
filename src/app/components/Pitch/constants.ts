interface IPitchFormationProps {
  [key: string]: object;
}

export const PITCH_FORMATIONS: IPitchFormationProps = {
  442: {
    tn: `"_1" "_2" "_5" "_6" "_3" "_7" "_4" "_8" "_11" "_9" "_10"`,
    '321px': `
      ". . . .       . . . .       _1 _1 _1 _1       . . . .       . . . ."

      "_2 _2 _2 _2 _2    _5 _5 _5 _5 _5    _6 _6 _6 _6 _6    _3 _3 _3 _3 _3"

      "_7 _7 _7 _7 _7    _4 _4 _4 _4 _4    _8 _8 _8 _8 _8    _11 _11 _11 _11 _11"

      ". . . . .      _9 _9 _9 _9 _9      _10 _10 _10 _10 _10      . . . . ."

    `
  },
  4411: {
    tn: `"_1" "_2" "_5" "_6" "_3" "_7" "_4" "_8" "_11" "_10" "_9"`,
    '321px': `
      ". . . .       . . . .       _1 _1 _1 _1       . . . .       . . . ."

      "_2 _2 _2 _2 _2    _5 _5 _5 _5 _5    _6 _6 _6 _6 _6    _3 _3 _3 _3 _3"

      "_7 _7 _7 _7 _7    _4 _4 _4 _4 _4    _8 _8 _8 _8 _8    _11 _11 _11 _11 _11"
      
      ". . . .      . . . .      _10 _10 _10 _10      . . . .       . . . ."

      ". . . .       . . . .       _9 _9 _9 _9       . . . .       . . . ."

    `
  },
  3421: {
    tn: `"_1" "_6" "_5" "_4" "_2" "_7" "_8" "_3" "_10" "_11" "_9"`,
    '321px': `
      ". . . .      . . . .        _1 _1 _1 _1       . . . .       . . . ."

      ". . . .     _6 _6 _6 _6     _5 _5 _5 _5     _4 _4 _4 _4     . . . ."

      "_2 _2 _2 _2 _2    _7 _7 _7 _7 _7    _8 _8 _8 _8 _8    _3 _3 _3 _3 _3"

      ". . . . .    _10 _10 _10 _10 _10    _11 _11 _11 _11 _11    . . . . ."

      ". . . .      . . . .        _9 _9 _9 _9       . . . .       . . . ."

    `
  }
}
