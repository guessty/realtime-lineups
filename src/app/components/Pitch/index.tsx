import * as React from 'react'
import { Grid } from 'react-grid-flexbox'
//
import StyledPitch from './StyledPitch'
import { PITCH_FORMATIONS } from './constants'

interface IPitchProps {
  formationType: string
  children?: any
}

interface IChildProps {
  formationPlace: string | number
}

const Pitch = (props:IPitchProps) => {
  const PlacedChildren = React.Children.map(props.children, (child: React.ReactElement<IChildProps>) => {
    return (
      <div data-grid-area={`_${child.props.formationPlace}`}>
        {child}
      </div>
    )
  })
  return (
    <React.Fragment>
      <StyledPitch className="pitch">
        <Grid
          templateAreas={PITCH_FORMATIONS[props.formationType]}
          templateColumns={{
            tn: '1fr',
            '321px': 'repeat(20, 1fr)',
          }}
        >
          {PlacedChildren}
        </Grid>
      </StyledPitch>
    </React.Fragment>
  )
}

export default Pitch
