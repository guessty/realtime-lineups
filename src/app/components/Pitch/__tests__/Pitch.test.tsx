import * as React from 'react'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as renderer from 'react-test-renderer'
//
import Pitch from './../index'
// import { PITCH_FORMATIONS } from './../constants'

Enzyme.configure({adapter: new Adapter()})


describe('<Pitch />', () => {
  let component: any
  let componentWrapper: any
  const props = {
    formationType: '442'
  }

  beforeEach(() => {
    component = (
      <Pitch {...props} />
    )
    componentWrapper = Enzyme.shallow(component, {})
  })

  it('should be defined', () => {
    expect(Pitch).toBeDefined()
  })

  it('should render', () => {
    expect(componentWrapper.length).toEqual(1)
  })
  
  it('capturing Snapshot of Pitch', () => {
    const renderedValue =  renderer.create(component).toJSON()
    expect(renderedValue).toMatchSnapshot();
  })
})

