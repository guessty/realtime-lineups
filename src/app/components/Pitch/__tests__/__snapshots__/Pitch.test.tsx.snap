// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pitch /> capturing Snapshot of Pitch 1`] = `
<div
  className="pitch sc-ifAKCX bXqWpb"
>
  <div
    className="sc-bwzfXH hhzUwB"
  />
</div>
`;
